### PR TITLE
SAK-48323 webcomponents. Bundle and remove empathise.

### PR DESCRIPTION
**Pull request by adrianfish**
https://sakaiproject.atlassian.net/browse/SAK-48323

### CODE REVIEW

**Comment by jesusmmp** [comment]
@adrianfish @ern I think we should avoid merging big changes before
tickets from Unidigital were merged. Please, take in mind. Thanks!

**Comment by adrianfish** [comment]
@jesusmmp @mpellicer Agreed. That's why I've put the PR up early, for transparency.

**Comment by adrianfish** [comment]
@mpellicer Are there particular commits in the s2u branch that you are worried about? This PR mainly affects the web components, the lit stuff, so I'm hopeful we won't get a lot of overlap. Is there any chance you can prioritise merging the risky commits into master? Can you point me to those commits so I can take a look?

**Comment by adrianfish** [comment]
@mpellicer @bgarciaentornos
This PR's getting a bit painful with the conflicts and I'd like to get it in soonish, if possible. So, if we could get any webcomponents PRs into master soon for review, that would be great.

**Comment by mpellicer** [comment]
Well @adrianfish, we have very very limited resources to contribute stuff to Master right now, but we made the weekly meeting today and I've assigned @stetsche to contribute the most critical features, with the help of @bgarciaentornos.

We're going to contribute to Master this two tickets very soon:

https://sakaiproject.atlassian.net/browse/S2U-34
https://sakaiproject.atlassian.net/browse/S2U-5

@bgarciaentornos Do you feel that other tickets need contribution? I don't see any here:

https://github.com/sakaiproject/sakai/commits/s2u-22x/webcomponents

Thanks

**Comment by bgarciaentornos** [comment]
You're right, those 2 seem to be the only tickets with related changes (although I'm seeing here other backend changes on tools like assignments but I'm guessing that's because it's still a draft).

**Comment by adrianfish** [comment]
Great news, thanks for putting people on this. I know you're on a tight schedule.

**Comment by stetsche** [comment]
Hi @adrianfish,  I can not work on the migration of S2U-5 at the moment because of other tasks. It will take some time until I can get to it, so this PR can go in first, let's not block it further!

**Comment by adrianfish** [comment]
@stetsche Great news. I can stop rebasing it now then :)

I'll do a load of testing today and I'll ask you for a review. Obviously, it is a big PR, but if you could just skim over it that would be good.

File: library/src/webapp-filtered/editor/ckeditor.launch.js [hunk 1]
```diff
@@ -351,7 +357,7 @@ sakai.editor.editors.ckeditor.launch = function(targetId, config, w, h) {
         browserContextMenuOnCtrl: true,
         
         // Fix the smileys to a single location
-        smiley_path: `/library/webjars/ckeditor4/${ckeditor.version}/plugins/smiley/images/`,
+        smiley_path: `/library/webjars/ckeditor4/4.21.0/plugins/smiley/images/`,
```
Comment (stetsche) [inline library/src/webapp-filtered/editor/ckeditor.launch.js:360]: Should all those versions be hardcoded?

File: library/src/webapp/editor/ckextraplugins/audiorecorder/plugin.js [hunk 2]
```diff
@@ -184,19 +184,3 @@ Audio.prototype.setObjectElement = function (e){
 		this.width = GetAttribute( e, 'width', this.width );
 		this.height = GetAttribute( e, 'height', this.height );
 };
-
-String.prototype.endsWith = function(str)
```
Comment (stetsche) [inline library/src/webapp/editor/ckextraplugins/audiorecorder/plugin.js:188]: Are those safe to remove?

File: portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/site.vm [hunk 3]
```diff
@@ -27,6 +27,17 @@
 
     <body class="Mrphs-portalBody overflow-hidden $!{loggedInClass}" data-bs-no-jquery="true">
 
+        <script>
```
Comment (stetsche) [inline portal/portal-render-engine-impl/impl/src/webapp/vm/morpheus/site.vm:30]: Could this be a small script file? Then it can be cached. Else this needs to be loaded every page load?

File: samigo/samigo-app/src/webapp/jsf/author/item/rubricAssociation.jsp [hunk 4]
```diff
@@ -17,7 +17,7 @@
                 state-details='<h:outputText value="#{itemauthor.rubricStateDetails}"/>'
             </h:panelGroup>
```
Comment (stetsche) [inline samigo/samigo-app/src/webapp/jsf/author/item/rubricAssociation.jsp:18]: Not related to the PR, but I am just wondering if that is not rendering spans in here

File: velocity/tool/src/templates/VM_chef_library.vm [hunk 5]
```diff
@@ -542,7 +542,7 @@ ${extra}")
 *# ------------------------------------------------------------------
 #macro (chef_setupformattedtextarea $textarea_id)
 	#if (!$is_wireless_device)
-		<script type="text/javascript" defer="1">sakai.editor.launch('$textarea_id');</script>
```
Comment (stetsche) [inline velocity/tool/src/templates/VM_chef_library.vm:545]: is this change needed?

File: webcomponents/tool/src/main/frontend/packages/sakai-conversations/web-test-runner.config.mjs [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        <link rel="stylesheet" src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap/5.0.1/css/bootstrap.min.css" />
+        <script src="https://cdnjs.cloudflare.com/ajax/libs/bootstrap/5.0.1/js/bootstrap.bundle.min.js"></script>
+      </head>
+      <body>
+        <script>window.process = { env: { NODE_ENV: "development" } }</script>
+        <script type="module" src="${testFramework}"></script>
+      </body>
+    </html>`,
+
+  /** Compile JS for older browsers. Requires @web/dev-server-esbuild plugin */
```
Comment (stetsche) [inline webcomponents/tool/src/main/frontend/packages/sakai-conversations/web-test-runner.config.mjs:36]: Do we need this?

File: webcomponents/tool/src/main/frontend/packages/sakai-course-card/src/SakaiCourseCard.js [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+
+    #course-options { min-width: 200px; padding: 10px; }
+    #course-options input { margin-right: 10px; }
+      #favourite-block {
+        color: var(--sakai-course-card-options-menu-favourites-block-color);
+        font-size: var(--sakai-course-card-options-menu-favourites-block-font-size);
+        font-weight: var(--sakai-course-card-options-menu-favourites-block-font-weight);
+      }
+
+    /*
```
Comment (stetsche) [inline webcomponents/tool/src/main/frontend/packages/sakai-course-card/src/SakaiCourseCard.js:183]: Do we need this?

File: webcomponents/tool/src/main/frontend/packages/sakai-course-card/web-test-runner.config.mjs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+  filterBrowserLogs(log) {
+    for (const arg of log.args) {
+      if (typeof arg === 'string' && filteredLogs.some(l => arg.includes(l))) {
+        return false;
+      }
+    }
+    return true;
+  },
+
+  /** Compile JS for older browsers. Requires @web/dev-server-esbuild plugin */
```
Comment (stetsche) [inline webcomponents/tool/src/main/frontend/packages/sakai-course-card/web-test-runner.config.mjs:24]: Do we need this?

File: webcomponents/tool/src/main/frontend/packages/sakai-course-dashboard/src/SakaiCourseDashboard.js [hunk 9]
```diff
@@ -53,7 +48,7 @@ export class SakaiCourseDashboard extends LitElement {
         throw new Error(`Failed to get dashboard data from ${url}`);
 
       })
-      .then(r => this.data = r)
+      .then(r => { console.log(r); this.data = r; })
```
Comment (stetsche) [inline webcomponents/tool/src/main/frontend/packages/sakai-course-dashboard/src/SakaiCourseDashboard.js:51]: console.log :)

**Review by stetsche — CHANGES_REQUESTED** [review]
I looked mostly through the code outside of webcomponents and added a few comments there. I saw some changes related to loadProperties and take it, as it is available globally now, what bundle is including that?
As for the changes in webcomponents, it's a bit too much to review for me and I am not sure what is new and what is moved. Though I noticed there is a lot of commented out code on the tests. There are also some codacy issues that you should be able to address without without risk. And I am wondering if you could put the rule to put arrow functions in parenthesis into the eslint config (warn), because codacy will complain about it anyway. One more thing I am wondering is, if the sakai-ui components are still here or in an external repository.
It's hard to get an overview with so much changes, but overall this seems to bring in more structure into the webcomponents build process, good job with that!

File: assignment/tool/src/webapp/vm/assignment/instructor_new_edit/grading_section.vm [hunk 10]
```diff
@@ -53,27 +53,24 @@ This section contains the options for grading this assignment, including rubrics
         ## Rubrics
         <div class="form-group">
             <sakai-rubric-association
-                site-id="$siteId"
+                site-id="0a480221-f3ce-4865-a952-12cb69fab076"
```
[the remark this answers is not in the document]
Comment (adrianfish) [reply assignment/tool/src/webapp/vm/assignment/instructor_new_edit/grading_section.vm:56]: Thanks. Well spotted.

File: assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java [hunk 11]
```diff
@@ -1229,7 +1230,8 @@ public ActionReturn setGrade(Map<String, Object> params) {
         if (submission != null) {
             boolean anonymousGrading = assignmentService.assignmentUsesAnonymousGrading(assignment);
             try {
-                return new ActionReturn(submissionToMap(activeSubmitters, assignment, new SimpleAssignment(assignment),  submission, true));
+                return new ActionReturn(Map.of("assignmentCloseTime", assignment.getCloseDate(),
```
[the remark this answers is not in the document]
Comment (adrianfish) [reply assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java:1233]: I'm not actually sure. I think maybe the data being returned was just too verbose for a saved grade. I will check my reasoning there.

File: assignment/tool/src/webapp/vm/assignment/chef_assignments_instructor_new_edit_assignment.vm [hunk 12]
```diff
@@ -186,7 +187,7 @@
                         <input type="hidden" id="$entitykey" name="$entitykey" value="$entityval">
 
 				<table border="0" style="margin:0"><tr><td>
-					<textarea name="$name_Description" id="$name_Description" cols="80" rows="30" wrap="virtual">$formattedText.escapeHtmlFormattedTextarea($value_Description)</textarea>
+					<textarea name="$name_Description" id="new_assignment_instructions" cols="80" rows="30" wrap="virtual">$formattedText.escapeHtmlFormattedTextarea($value_Description)</textarea>
```
[the remark this answers is not in the document]
Comment (adrianfish) [reply assignment/tool/src/webapp/vm/assignment/chef_assignments_instructor_new_edit_assignment.vm]: Not sure. I've reverted that change.

File: webcomponents/tool/src/main/frontend/packages/sakai-conversations/web-test-runner.config.mjs (hunk 6 above)
Comment (adrianfish) [reply webcomponents/tool/src/main/frontend/packages/sakai-conversations/web-test-runner.config.mjs:36]: We need the test runner file, but if you mean the commented stuff, I just leave that in for doc purposes.

File: velocity/tool/src/templates/VM_chef_library.vm (hunk 5 above)
Comment (adrianfish) [reply velocity/tool/src/templates/VM_chef_library.vm:545]: Yes. Defer does nothing in this case as it only works on actual sources script loads. This is just an inline script which calls a function.

File: assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java (hunk 11 above)
[the remark this answers is not in the document]
Comment (adrianfish) [reply assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java:1233]: It is needed. I needed to get assignmentCloseTime returned so it could be updated on the current submission object. However, I may rethink that a little. I think we don't need all that data sending back on the network.

File: commons/tool/src/webapp/js/commons.js [hunk 13]
```diff
@@ -374,14 +374,11 @@ commons.switchState = function (state, arg) {
 
     $(document).ready(function () {
 
-      import("/webcomponents/sakai-i18n.js").then(m => {
+      loadProperties({bundle: 'commons'}).then(i18n => {
```
[the remark this answers is not in the document]
Comment (adrianfish) [reply commons/tool/src/webapp/js/commons.js:377]: I really don't want to do that. We should not have to clutter up single parameter function calls with unneeded parentheses. I'll look at updating our Codacy rules instead. I ignore Codacy on JS, if I am to be honest. Eslint and lit-analyzer are much more useful.

File: assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java (hunk 11 above)
[the remark this answers is not in the document]
Comment (adrianfish) [reply assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java:1233]: No, it's needed as it is. Previously, the assignmentCloseTime was being serialised with each submission and bloating the json unnecessarily, so I moved it out of the submissionMap.

[the remark this answers is not in the document]
Comment (stetsche) [reply assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java:1233]: Allright!

File: commons/tool/src/webapp/js/commons.js (hunk 13 above)
[the remark this answers is not in the document]
Comment (stetsche) [reply commons/tool/src/webapp/js/commons.js:377]: I think codacy will also just run something like eslint, I think it would be good if the rules are matching.

File: webcomponents/tool/src/main/frontend/packages/sakai-conversations/web-test-runner.config.mjs (hunk 6 above)
Comment (stetsche) [reply webcomponents/tool/src/main/frontend/packages/sakai-conversations/web-test-runner.config.mjs:36]: Yes, it's about the commented code, if it's good to keep it, i'm fine with it.

File: commons/tool/src/webapp/js/commons.js (hunk 13 above)
[the remark this answers is not in the document]
Comment (stetsche) [reply commons/tool/src/webapp/js/commons.js:377]: I see you are on to it ;)

[the remark this answers is not in the document]
Comment (adrianfish) [reply commons/tool/src/webapp/js/commons.js:377]: I agree they should match. What I don't want to do though is force the webcomponents rules on every piece of js in Sakai. That would be chaos. I'll read up on codacy a bit - you would think that codacy would be smart enough to spot eslint config files in sub directories.

File: library/src/webapp/editor/ckextraplugins/audiorecorder/plugin.js (hunk 2 above)
Comment (adrianfish) [reply library/src/webapp/editor/ckextraplugins/audiorecorder/plugin.js:188]: yes. endsWith is now on String OOTB, and it does the same thing.

File: library/src/webapp-filtered/editor/ckeditor.launch.js (hunk 1 above)
Comment (adrianfish) [reply library/src/webapp-filtered/editor/ckeditor.launch.js]: No! Thanks for spotting that. No idea what was going on there.

File: library/src/webapp/editor/ckextraplugins/audiorecorder/plugin.js (hunk 2 above)
Comment (stetsche) [reply library/src/webapp/editor/ckextraplugins/audiorecorder/plugin.js:188]: nice :+1:

Comment (adrianfish) [reply library/src/webapp/editor/ckextraplugins/audiorecorder/plugin.js:188]: Also, it overwrites the endsWith for any other js on the page at the time :)

**Comment by adrianfish** [comment]
@stetsche Thanks for the review, btw. Really appreciate it. You spotted some real crap from me :)

**Comment by adrianfish** [comment]
@stetsche The ui components are staying in the main Sakai repo for now. It seemed to make sense to just change things gradually. We can get the benefits of a monorepo approach from inside the webcomponents module. We can publish components to the npm repo, all that stuff, but the code still stays together. That may change in the future, but I didn't want to bite too much off at once.

File: library/src/webapp-filtered/editor/ckeditor.launch.js [hunk 14]
```diff
@@ -467,19 +473,19 @@ sakai.editor.editors.ckeditor.launch = function(targetId, config, w, h) {
 
         //These could be applicable to the basic toolbar
         CKEDITOR.plugins.addExternal('lineutils',basePath+'lineutils/', 'plugin.js');
-        CKEDITOR.plugins.addExternal('html5video',webJars+'ckeditor-html5-video/${ckeditor.html5video.version}/', 'plugin.js');
+        CKEDITOR.plugins.addExternal('html5video',webJars+'ckeditor-html5-video/1.1.1/', 'plugin.js');
```
Comment (stetsche) [inline library/src/webapp-filtered/editor/ckeditor.launch.js:476]: Should those versions also be restored?

**Comment by stetsche** [comment]
@adrianfish You are welcome, thanks for the info!

File: library/src/webapp/editor/ckextraplugins/audiorecorder/plugin.js (hunk 2 above)
Comment (stetsche) [reply library/src/webapp/editor/ckextraplugins/audiorecorder/plugin.js:188]: Yes, that was my concern, but all good.

File: webcomponents/tool/src/main/frontend/packages/fa-icon/package.json [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+    "src"
+  ],
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/sakaiproject/sakai-ui.git"
+  },
+  "keywords": [
+    "fontawesome"
+  ],
+  "author": "The Sakai Project <sakai.user.interface@gmail.com>",
```
Comment (ern) [inline webcomponents/tool/src/main/frontend/packages/fa-icon/package.json:23]:
```suggestion
  "author": "The Sakai Project (https://sakailms.org)",
```

**Comment by ern** [comment]
Adrian so this will all end up in the sakai-ui repo? and I am guessing this is the a step before that happens? If so I think it would be a good idea to update the webcomponents strategy doc and layout the direction there?

this is very good direction so that we can start publishing webcomponents separately from Sakai, nice!!!

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/TasksController.java [hunk 16]
```diff
@@ -137,38 +132,24 @@ public Map<String, Object> getSiteTasks(@PathVariable String siteId) throws User
...
-            Site site = siteService.getSite(siteId);
             // Returns a boolean value which depends if an user is an instructor or not
-            return securityService.unlock(SiteService.SECURE_UPDATE_SITE, site.getReference());
+            if (securityService.unlock(SiteService.SECURE_UPDATE_SITE, site.getReference())) {
+                data.put("canUpdateSite", true);
+
+                Collection<Group> groups = site.getGroups();
+                if (!groups.isEmpty()) {
+                    Map<String, String> groupsMap = (Map<String, String>) groups.stream().collect(Collectors.toMap(Group::getId, Group::getTitle));
+                    data.put("groups", groupsMap.entrySet().toArray());
```
Comment (stetsche) [inline webapi/src/main/java/org/sakaiproject/webapi/controllers/TasksController.java:143]: Why do we need the map? If it needs to be an array, could it be something like `stream().map(Group::getTitle).toArray(String[]::new)`? Or could we rather collect the titles to a Set?

File: webcomponents/tool/src/main/frontend/packages/fa-icon/package.json (hunk 15 above)
Comment (stetsche) [reply webcomponents/tool/src/main/frontend/packages/fa-icon/package.json:23]: I think this makes sense

File: webcomponents/tool/src/main/frontend/packages/sakai-announcements/test/sakai-announcements.test.js [hunk 17]
```diff
@@ -0,0 +1,106 @@
...
+    expect(el.shadowRoot.getElementById("topbar")).to.exist;
+    expect(el.shadowRoot.querySelectorAll("div.title").length).to.equal(2);
+
+    expect(el.shadowRoot.querySelectorAll("div.announcements > .header").length).to.equal(2);
+
+    //el.setAttribute("site-id", data.vavavoom);
+
+    //await el.updateComplete;
+
+    //expect(el.shadowRoot.querySelectorAll("div.title").length).to.equal(1);
```
Comment (stetsche) [inline webcomponents/tool/src/main/frontend/packages/sakai-announcements/test/sakai-announcements.test.js:93]: Those might be to remove?

File: webcomponents/tool/src/main/frontend/packages/sakai-conversations/src/SakaiTopic.js [hunk 18]
```diff
@@ -734,13 +732,13 @@ export class SakaiTopic extends reactionsMixin(SakaiElement) {
                     @click=${this.toggleReaction}
                     aria-label="${this.topic.myReactions.GOOD_QUESTION ? this._i18n.ungoodquestion_tooltip : this._i18n.goodquestion_tooltip}"
                     title="${this.topic.myReactions.GOOD_QUESTION ? this._i18n.ungoodquestion_tooltip : this._i18n.goodquestion_tooltip}">
-                  ${this._i18n.goodquestion} ${this.isInstructor && this.topic.reactionTotals.GOOD_QUESTION ? ` - ${this.topic.reactionTotals.GOOD_QUESTION}` : ""}
+                  ${this._i18n.goodquestion} ${this.isInstructor && this.topic.reactionTotals.GOOD_QUESTION ? ` - ${this.topic.reactionTotals.GOOD_QUESTION}` : "" }
```
Comment (stetsche) [inline webcomponents/tool/src/main/frontend/packages/sakai-conversations/src/SakaiTopic.js:735]: Not needed I think

File: webcomponents/tool/src/main/frontend/packages/sakai-conversations/src/SakaiTopicSummary.js [hunk 19]
```diff
@@ -35,8 +30,8 @@ export class SakaiTopicSummary extends SakaiElement {
     return html`
       <a href="javascript:;" @click=${this._topicSelected} class="topic-summary-link">
         <div class="topic-summary
-              ${this.topic.numberOfUnreadPosts > 0 && !this.topic.selected ? " unread" : ""}
-              ${this.topic.selected ? "selected" : ""}"
+              ${this.topic.numberOfUnreadPosts > 0 && !this.topic.selected ? " unread" : "" }
+              ${this.topic.selected ? "selected" : "" }"
```
Comment (stetsche) [inline webcomponents/tool/src/main/frontend/packages/sakai-conversations/src/SakaiTopicSummary.js:34]: extra spaces

File: webcomponents/tool/src/main/frontend/packages/sakai-editor/LICENSE [hunk 20]
```diff
@@ -0,0 +1,12 @@
+Copyright 2022 The Apereo Foundation Licensed under the
```
Comment (stetsche) [inline webcomponents/tool/src/main/frontend/packages/sakai-editor/LICENSE:1]: Should this be 2024?

File: webcomponents/tool/src/main/frontend/packages/sakai-rubrics/src/SakaiRubricsHelpers.js [hunk 21]
```diff
@@ -3,7 +3,7 @@ export class SakaiRubricsHelpers {
   static handleErrors(response) {
 
     if (!response.ok) {
-      console.error(`Error : ${  response.statusText || response.status}`);
+      console.error(`Error : ${ response.statusText || response.status}`);
```
Comment (stetsche) [inline webcomponents/tool/src/main/frontend/packages/sakai-rubrics/src/SakaiRubricsHelpers.js:6]: Maybe one more space to remove

**Review by stetsche — COMMENTED** [review]
I looked over the PR again and added a few comments, I also saw some other unresolved comments. Again, I didn't go through every change in webcomponents, but I looked over a few.

File: webapi/src/main/java/org/sakaiproject/webapi/controllers/TasksController.java (hunk 16 above)
Comment (adrianfish) [reply webapi/src/main/java/org/sakaiproject/webapi/controllers/TasksController.java]: Good spot! The whole groups json was broken. I've fixed the tasks tests and made sure it works in production.

**Comment by adrianfish** [comment]
Regarding the package author entries:
https://docs.npmjs.com/cli/v10/configuring-npm/package-json#people-fields-author-contributors

It's recommended to supply the email address and the one I've used is attached to a sakai-ui npm repository at https://www.npmjs.com/~sakai-ui. At the moment we are not publishing to it as we are linking between npm modules in our bundling, but we will be at some point.

**Comment by adrianfish** [comment]
The problem is that we already have the sakai-ui namespace on npm, and the login is sakai.user.interface@gmail.com. We could use the expanded multiple lines version, but the single line conforms with the way git histories show things.

File: webcomponents/tool/src/main/frontend/packages/sakai-editor/LICENSE (hunk 20 above)
Comment (adrianfish) [reply webcomponents/tool/src/main/frontend/packages/sakai-editor/LICENSE:1]: I'm not actually sure. The code was not written this year.

File: library/src/webapp/js/sakai-message-broker.js [hunk 22]
```diff
@@ -3,6 +3,14 @@ portal.notifications = portal.notifications || {};
 
 portal.notifications.pushCallbacks = new Map();
 
+portal.notifications.clearAppBadge = () => {
```
[the remark this answers is not in the document]
Comment (adrianfish) [reply library/src/webapp/js/sakai-message-broker.js:6]: It will clear any number in the app badge, so the icon in the osx toolbar, or your phone

[the remark this answers is not in the document]
Comment (stetsche) [reply library/src/webapp/js/sakai-message-broker.js:6]: But it's just an `console.debug`

[the remark this answers is not in the document]
Comment (adrianfish) [reply library/src/webapp/js/sakai-message-broker.js:6]: Ah, right. That's different then :) I'll take a look at that now. It may be leakage from the pwa stuff I've been working on.

**Review by stetsche — APPROVED** [review]
